packages/monaco-editor-react/src/index.tsx

```diff
@@ -67,76 +91,85 @@ export class MonacoEditorReactComp<T extends MonacoEditorProps = MonacoEditorPro
             mustReInit = (wrapper?.getMonacoEditorApp() as EditorAppExtended).isAppConfigDifferent(prevConfig, config, false) === true;
         }
 
-        if (mustReInit) {
-            await this.handleReinit();
-        } else {
-            // the function now ensure a model update is only required if something else than the code changed
-            this.wrapper.updateModel(userConfig.wrapperConfig.editorAppConfig);
-
-            if (prevConfig.$type === 'classic' && config.$type === 'classic') {
-                if (prevConfig.editorOptions !== config.editorOptions) {
-                    (wrapper.getMonacoEditorApp() as EditorAppClassic).updateMonacoEditorOptions(config.editorOptions ?? {});
-                }
-            }
-        }
+        return mustReInit;
     }
 
     override componentWillUnmount() {
         this.destroyMonaco();
     }
 
-    private assignRef = (component: HTMLDivElement) => {
+    protected assignRef = (component: HTMLDivElement) => {
         this.containerElement = component;
     };
 
-    private async destroyMonaco(): Promise<void> {
+    protected async destroyMonaco(): Promise<void> {
         if (this.wrapper) {
-            await this.isStarting;
+            if (this.isRestaring) {
+                await this.isRestaring;
+            }
             try {
                 await this.wrapper.dispose();
             } catch {
-                // This is fine
-                // Sometimes the language client throws an error during disposal
-                // This should not prevent us from continue working
+                // The language client may throw an error during disposal.
+                // This should not prevent us from continue working.
             }
         }
         if (this._subscription) {
             this._subscription.dispose();
         }
     }
 
-    private async initMonaco() {
+    protected async initMonaco() {
+        const {
+            userConfig
+        } = this.props;
+
+        // block "destroyMonaco" until start is complete
+        this.isRestaring = new Promise<void>((resolve) => {
+            this.started = resolve;
+        });
+        await this.wrapper.init(userConfig);
+    }
+
+    protected async startMonaco() {
         const {
             className,
-            userConfig,
-            onTextChanged,
             onLoad,
         } = this.props;
 
         if (this.containerElement) {
             this.containerElement.className = className ?? '';
 
-            this.isStarting = this.wrapper.start(userConfig, this.containerElement);
-            await this.isStarting;
+            await this.wrapper.startNoInit(this.containerElement);
+            this.started();
+            this.isRestaring = undefined;
```

Comment:
Small typo, 'isRestarting'

Reply:
Thank you. Will be resolved in with next commit.